exporter/awsemfexporter/datapoint.go

```diff
@@ -213,42 +235,122 @@ func (dps histogramDataPointSlice) IsStaleNaNInf(i int) (bool, pcommon.Map) {
 }
 
 // CalculateDeltaDatapoints retrieves the ExponentialHistogramDataPoint at the given index.
+// As CloudWatch EMF logs allows in maximum of 100 target members, the exponential histogram metric are split into two data points if the total buckets exceed 100,
+// the first data point contains the first 100 buckets, while the second data point includes the remaining buckets.
+// Re-calculated Min, Max, Sum, Count for each split:
+// 1. First split datapoint:
+//   - Max: From original metric.
+//   - Min: Last bucket’s bucketBegin in the first split.
+//   - Sum: From original metric.
+//   - Count: Calculated from the first split buckets.
+//
+// 2. Second split datapoint:
+//   - Max: First bucket’s bucketEnd in the second split.
+//   - Min: From original metric.
+//   - Sum: 0.
+//   - Count: Overall count - first split count.
 func (dps exponentialHistogramDataPointSlice) CalculateDeltaDatapoints(idx int, _ string, _ bool, _ *emfCalculators) ([]dataPoint, bool) {
 	metric := dps.ExponentialHistogramDataPointSlice.At(idx)
-
 	scale := metric.Scale()
 	base := math.Pow(2, math.Pow(2, float64(-scale)))
-	arrayValues := []float64{}
-	arrayCounts := []float64{}
+
 	var bucketBegin float64
 	var bucketEnd float64
+	totalBucketLen := metric.Positive().BucketCounts().Len() + metric.Negative().BucketCounts().Len()
+	if metric.ZeroCount() > 0 {
+		totalBucketLen++
+	}
+	firstSplit := dataPointSplit{
+		values:     []float64{},
+		counts:     []float64{},
+		count:      int(metric.Count()),
+		length:     0,
+		bucketMin:  metric.Min(),
+		bucketMax:  metric.Max(),
+		maxBuckets: totalBucketLen,
+	}
+
+	currentLength := 0
+	splitThreshold := 100
+	var secondSplit dataPointSplit
+	if totalBucketLen > splitThreshold {
+		firstSplit.maxBuckets = splitThreshold
+		firstSplit.count = 0
+		secondSplit = dataPointSplit{
+			values:     []float64{},
+			counts:     []float64{},
+			count:      0,
+			length:     0,
+			bucketMin:  metric.Min(),
+			bucketMax:  metric.Max(),
+			maxBuckets: totalBucketLen - splitThreshold,
+		}
+	}
 
 	// Set mid-point of positive buckets in values/counts array.
 	positiveBuckets := metric.Positive()
 	positiveOffset := positiveBuckets.Offset()
 	positiveBucketCounts := positiveBuckets.BucketCounts()
 	bucketBegin = 0
 	bucketEnd = 0
-	for i := 0; i < positiveBucketCounts.Len(); i++ {
+	for i := positiveBucketCounts.Len() - 1; i >= 0; i-- {
```

Comment:
Did we just reverse this so the max value is in the first split?

Reply:
Yes,  the positiveBucket range is from low to high, I'm reverting it so that we can always assign higher values to front datapoints